src/routers/upload.py

```diff
@@ -41,19 +45,142 @@ def format_size(size: int) -> str:
         return f"{size / 1024**3:.2f} GB"
 
 
+async def combine_chunks(temp_dir: Path, total_chunks: int, filename: str) -> Path:
+    """Combine all chunks into a single file."""
+    combined_file = temp_dir / filename
+    with combined_file.open("wb") as outfile:
+        for i in range(int(total_chunks)):
+            chunk_file = temp_dir / f"chunk_{i}"
+            with chunk_file.open("rb") as infile:
+                outfile.write(infile.read())
+    return combined_file
+
+
+def compute_sha256(file_path: Path) -> str:
+    """Compute SHA256 checksum of a file."""
+    sha256_hash = hashlib.sha256()
+    with file_path.open("rb") as f:
+        for byte_block in iter(lambda: f.read(4096), b""):
+            sha256_hash.update(byte_block)
+    return sha256_hash.hexdigest()
+
+
+def create_dataset_entry(
+    filename: str, target_path: Path, sha256: str, bounds, user_id: str, copy_time: int
+) -> Dataset:
+    """Create a new dataset entry in the database."""
+    data = dict(
+        file_name=target_path.name,
+        file_alias=filename,
+        file_size=target_path.stat().st_size,
+        sha256=sha256,
+        bbox=bounds,
+        status=StatusEnum.pending,
+        user_id=user_id,
+        copy_time=copy_time,
+    )
+    dataset = Dataset(**data)
+
+    with use_client() as client:
+        try:
+            send_data = {
+                k: v
+                for k, v in dataset.model_dump().items()
+                if k != "id" and v is not None
+            }
+            response = client.table(settings.datasets_table).insert(send_data).execute()
+        except Exception as e:
+            logger.exception(f"Error uploading dataset: {str(e)}")
+            raise HTTPException(
+                status_code=400, detail=f"Error uploading dataset: {str(e)}"
+            )
+
+    return Dataset(**response.data[0])
+
+
+@router.post("/datasets/chunk")
+async def upload_geotiff_chunk(
+    file: UploadFile,
+    chunk: Annotated[int, Form()],
+    chunks: Annotated[int, Form()],
+    filename: Annotated[str, Form()],
+    copy_time: Annotated[int, Form()],
+    upload_id: Annotated[str, Form()],
+    token: Annotated[str, Depends(oauth2_scheme)],
+):
+    """
+    Handle chunked upload of a GeoTIFF file.
+
+    This endpoint receives chunks of a file, saves them temporarily,
+    and combines them when all chunks are received.
+    """
+    # Verify the token
+    user = verify_token(token)
+    if not user:
+        raise HTTPException(status_code=401, detail="Invalid token")
+
+    # Define the path for temporary chunk storage
+    temp_dir = Path(settings.tmp_upload_path) / upload_id
+    temp_dir.mkdir(parents=True, exist_ok=True)
+
+    # Save the chunk
+    chunk_file = temp_dir / f"chunk_{chunk}"
```

Comment:
Do I see this correctly that the filenames are simply chunk_{someid}? Do parallel uploads from different (or the same) users work with this concept? I.e, how do you know which chunk belongs to which file?

Reply:
The `tmp_dir` is set by `temp_dir = Path(settings.tmp_upload_path) / upload_id` 
and `upload_id` comes from the frontend and is a unique key for each upload, which is only used to handle the upload-based notifications. 
So it's a unique identifier.

Reply:
Ah I see. So tmp_dir != temp_dir ?

Reply:
Maybe the variable name should be changed...